src/Server.php

```diff
@@ -549,17 +573,17 @@ protected function getSmdMethods()
     /**
      * Internal method for handling request.
      *
-     * @return void
+     * @return Error|null
      */
-    protected function handleRequest()
+    protected function handleRequest(): ?Error
     {
         $request = $this->getRequest();
 
         if ($request->isParseError()) {
             return $this->fault('Parse error', Error::ERROR_PARSE);
         }
 
-        if (! $request->isMethodError() && null === $request->getMethod()) {
+        if (! $request->isMethodError() && '' === $request->getMethod()) {
```

Comment:
@Xerkus Just so I get it right: Instead of an empty string strict comparison `$variable === ''`, `empty` construct should be used `empty($variable)` in general?

Reply:
I will not be asking you to change it to `empty()`.

There is a bit of a conundrum here:
- `empty()` does some special form of type juggling. That is bad. Eg `'0'` is considered empty.
- `=== ''` condition validity is not immediately clear during code review, in this case without checking return type declaration.
- value type not enforced by type declaration or type check must be treated as uncertain and potentially nullable.
- static analysis will spot `null === $request->getMethod()` as always false and will complain. But it will not notice `'' === $request->getMethod()` not covering `null`
- What makes `empty()` more clear is that it covers `null === $request->getMethod() || '' === $request->getMethod()`. It works regardless of value nullability and as such does not open a way for hard to spot logic error. 
- `empty()` treating certain values as empty opens a way for hard to spot logic errors when those values can appear among valid values.

Reply:
ACK